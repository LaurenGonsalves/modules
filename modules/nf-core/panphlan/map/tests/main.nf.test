nextflow_process {

    name "Test Process PANPHLAN_MAP"
    script "modules/nf-core/panphlan/map/main.nf"
    process "PANPHLAN_MAP"

    test("Should run without failures") {

            setup {
                run("PANPHLAN_DOWNLOADPANGENOME") {
                    script "modules/nf-core/panphlan/map/main.nf"
                    process {
                        """
                        input[0] = "Bacteroides_fragilis"
                        """
                    }
                }
            }

        when {
            process {
                """
                input[0] = PANPHLAN_DOWNLOADPANGENOME.out.pangenome
                input[1] = Channel.of (
                    [
                    PANPHLAN_DOWNLOADPANGENOME.out.indexes
                    ]
                )
                input[2] = Channel.of (
                    [
                    [id:"test1_1", single_end:true ],
                            file("https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz")
                    ]
                )
                input[3] = "Bacteroides_fragilis"


                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}