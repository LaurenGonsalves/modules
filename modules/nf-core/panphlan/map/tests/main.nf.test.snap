{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1_1",
                            "single_end": true
                        },
                        "test1_1_Bacteroides_fragilis.tsv:md5,6ac420a56792567619a08bc1d054706a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d9e8b34830d477d673900d100ea62420"
                ],
                "mapping_file": [
                    [
                        {
                            "id": "test1_1",
                            "single_end": true
                        },
                        "test1_1_Bacteroides_fragilis.tsv:md5,6ac420a56792567619a08bc1d054706a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d9e8b34830d477d673900d100ea62420"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-09T22:40:14.787830345"
    }
}